### ISSUE TITLE
no-shadowed-elements false-positive

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#441]
We have something like:

```hbs
<video class="video-js" preload="none" poster={{url model.poster}}>
  {{#each subtitles as |track index|}}
    <track src={{url track.id}} label={{track.name}} default={{eq index 0}}>
  {{/each}}
</video>
```

Which gives a `no-shadowed-elements` warning for the `track` element which, I think, is not correct (as far as I understand what the `no-shadowed-elements` rule does).

**Comment by rwjblue** [comment]
The rule is correct actually. In Ember 3.4 that exact code would attempt to invoke the `track` block param (which seems like it would error since from context it looks like a model?).

**Comment by rwjblue** [comment]
A bit more context here: https://github.com/rwjblue/ember-angle-bracket-invocation-polyfill/issues/17#issuecomment-397130893

**Comment by boris-petrov** [comment]
Ah, I see now. Thanks a bunch! It took me a while. :) Thanks for the great work on template-lint! And please be sure to write a lot of documentation and information on the new changes coming in Ember as they seem massive. :) Thanks again!

**Comment by frederikbosch** [comment]
I would not be surprised if many more people find this difficult, therefore a little more context. Do give you handlebars variables the same names as tags.

```hbs
{{#each list as |option|}}
 <option>this errors anyhow</option>
{{/each}}
```

Rather do.

```hbs
{{#each list as |optionData|}}
 <option>this does not error</option>
{{/each}}
```
